src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -200,8 +203,10 @@ def __init__(
         decoder: Optional[Union["BeamSearchDecoderCTC", str]] = None,
         device: Union[int, "torch.device"] = None,
         torch_dtype: Optional[Union[str, "torch.dtype"]] = None,
+        processor: Optional[AutoProcessor] = None,
         **kwargs,
     ):
+        self.processor = processor
```

Comment:
Unfortunately, we can't accept the `processor` as an attribute of the `pipeline`, for reasons mentioned [here](https://github.com/huggingface/transformers/pull/24952#pullrequestreview-1540567179).